artifactory/commands/python/poetry.go

```diff
@@ -95,15 +98,190 @@ func (pc *PoetryCommand) install(buildConfiguration *buildUtils.BuildConfigurati
+
+	// Get build name and number
+	buildName, err := buildConfiguration.GetBuildName()
 	if err != nil {
 		return err
 	}
-	// Run the publish cmd
+	buildNumber, err := buildConfiguration.GetBuildNumber()
+	if err != nil {
+		return err
+	}
+
+	// Get current working directory
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// Use FlexPack to collect dependencies with checksums
+	if buildName != "" && buildNumber != "" {
+		log.Info("Using FlexPack to collect Poetry dependencies...")
+
+		// Create FlexPack Poetry configuration
+		config := flexpack.PoetryConfig{
+			WorkingDirectory:       workingDir,
+			IncludeDevDependencies: false, // Match standard behavior
+		}
+
+		// Create Poetry FlexPack instance
+		poetryFlex, err := flexpack.NewPoetryFlexPack(config)
+		if err != nil {
+			log.Warn("Failed to create Poetry FlexPack, falling back to standard method: " + err.Error())
+		} else {
+			// Collect build info using FlexPack
+			flexBuildInfo, err := poetryFlex.CollectBuildInfo(buildName, buildNumber)
+			if err != nil {
+				log.Warn("Failed to collect build info with FlexPack, falling back to standard method: " + err.Error())
+			} else {
+				// Save FlexPack build info to be picked up by rt bp
+				err = pc.saveFlexPackBuildInfo(flexBuildInfo)
+				if err != nil {
+					log.Warn("Failed to save FlexPack build info: " + err.Error())
+				} else {
+					log.Info("Successfully saved FlexPack build info with dependencies")
+				}
+			}
+		}
+	}
+
+	// Run the publish command to upload artifacts
 	pc.args = publishCmdArgs
-	return gofrogcmd.RunCmd(pc)
+	err = gofrogcmd.RunCmd(pc)
+	if err != nil {
+		return err
+	}
+
+	// After successful publish, collect artifacts information
+	if buildName != "" && buildNumber != "" {
+		err = pc.collectPublishedArtifacts(buildConfiguration, pythonBuildInfo, workingDir)
+		if err != nil {
+			log.Warn("Failed to collect published artifacts: " + err.Error())
+		}
+	}
+
+	return nil
+}
+
+// saveFlexPackBuildInfo saves FlexPack build info for jfrog-cli rt bp compatibility
+func (pc *PoetryCommand) saveFlexPackBuildInfo(buildInfo *entities.BuildInfo) error {
+	// Create build-info service
+	service := build.NewBuildInfoService()
+
+	// Create or get build
+	bld, err := service.GetOrCreateBuildWithProject(buildInfo.Name, buildInfo.Number, "")
+	if err != nil {
+		return fmt.Errorf("failed to create build: %w", err)
+	}
+
+	// Save the complete build info (this will be loaded by rt bp)
+	err = bld.SaveBuildInfo(buildInfo)
+	if err != nil {
+		return fmt.Errorf("failed to save build info: %w", err)
+	}
+
+	return nil
+}
+
+// collectPublishedArtifacts collects information about artifacts that were published
+func (pc *PoetryCommand) collectPublishedArtifacts(buildConfiguration *buildUtils.BuildConfiguration, pythonBuildInfo *build.Build, workingDir string) error {
+	// Look for built artifacts in dist/ directory
+	distDir := filepath.Join(workingDir, "dist")
```

Comment:
can poetry write published artifacts to a different directory instead of default?